### PR TITLE
fixed reference before assignment bug with EMAIL variable

### PR DESCRIPTION
**Pull request by mattjmorgan**

### CODE REVIEW

**Comment by Ricardo-Osorio** [comment]
After checking out your pr, it seems like you forgot to add the import for "re"

File: main.py [hunk 1]
```diff
@@ -200,18 +211,12 @@ def execute():
 def main():
     global logger
     logger = logging.getLogger('egs-weekly-freegames')
-    read_env_variables()
+    if read_env_variables() is False:
```
Comment (Ricardo-Osorio) [inline main.py:249]: This will not enter the IF even when the email is invalid

File: main.py [hunk 2]
```diff
@@ -33,6 +33,17 @@ def read_env_variables():
     value = os.getenv('TOTP') or None
     TOTP = None if value is None else pyotp.TOTP(value)
 
+    if EMAIL:
+        try:
+            EMAIL = re.search("[^@ \t\r\n]+@[^@ \t\r\n]+\.[^@ \t\r\n]+", EMAIL)[0]
```
Comment (Ricardo-Osorio) [inline main.py:37]: you can add as many `.com` as you want to your email and it will still be valid with this expression

Comment (Ricardo-Osorio) [reply main.py:37]: It will also accept all kind of special characters.

Comment (Ricardo-Osorio) [reply main.py:37]: Perhaps its not even worth adding this kind of validation to the email. What are your thoughts?

**Review by Ricardo-Osorio — CHANGES_REQUESTED** [review]
Please take a look at the comments

File: main.py (hunk 2 above)
Comment (mattjmorgan) [reply main.py:37]: Odd the "re" was dropped. Still worked as I was testing.
As for if we need it, perhaps you're right.. This is an email validator that works "most of the time" (https://ihateregex.io/expr/email). Up to you really, I just added it to mine

Comment (Ricardo-Osorio) [reply main.py:37]: Happens 😄
I suggest dropping this change. Here's why. This is a small project and its execution it's meant to be automated, which means, after setting the trigger/cloudwatch event once (the email is set in this spet) you will not look at it again. For that reason I don't see a point to check the email on every execution.

Comment (mattjmorgan) [reply main.py:37]: Yea that's fine.. live and learn

**Pull request closed without merging**